### PR TITLE
[Feature/plugin] Zendesk integration 🚀

### LINKED ISSUE #3952: [Plugins] :  Zendesk integration
## Description
Integration with Zendesk's API to manage our tickets and resources efficiently. 
Use case:
Build internal tools like customer admin panels or analytics on ticket volumes and types, etc.

Feature Request: **ToolJet should connect to Zendesk to read and write data using Zendesk's OAuth**

Include queries:
```md
- List all tickets associated with an account
- List all users associated with an account
- List user-requested tickets
- Get ticket details(properties)
- Update a ticket
- Gets a user's profile with the given ID.
- Querying with Zendesk's Search API
```

### PR DESCRIPTION
**Pull request by arpitnath**
Resolves #3952

### CODE REVIEW

File: plugins/packages/zendesk/lib/index.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+          result = JSON.parse(response.body);
+          break;
+
+        case 'update_ticket':
+          response = await got(`${url}/tickets/${queryOptions['ticket_id']}.json`, {
+            method: 'put',
+            headers: this.authHeader(accessToken),
+            body: JSON.stringify(queryOptions['body']),
+          });
+          result = JSON.parse(response.body);
```
Comment (gsmithun4) [inline plugins/packages/zendesk/lib/index.ts:139]:
```suggestion
          result = await got(`${url}/tickets/${queryOptions['ticket_id']}.json`, {
            method: 'put',
            headers: this.authHeader(accessToken),
            json: queryOptions['body'],
          }).json();
```
`queryOptions['body']` is `string` or `json`? should parse if its `string`

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

File: plugins/packages/zendesk/lib/manifest.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    },
+    "client_secret": {
+      "value": ""
+    }
+  },
+  "properties": {
+    "zendesk": {
+      "label": "",
+      "key": "zendesk",
+      "type": "react-component-zendesk",
```
Comment (gondar00) [inline plugins/packages/zendesk/lib/manifest.json:37]: Q: would it not be possible to render this using dynamic form or alter dynamic form to in a certain way to handle zendesk's usecase?

Comment (arpitnath) [reply plugins/packages/zendesk/lib/manifest.json:37]: Yes. Let's do it in a separate PR and include gsheet and slack form.

**Review by gondar00 — APPROVED** [review]

**Comment by arpitnath** [comment]
Depends: #3069

**Comment by shubh22** [comment]
Task linked: [CU-2rqj138 Zendesk integration](https://app.clickup.com/t/2rqj138)

**Pull request merged**